homeassistant/components/sensor/zestimate.py

```diff
@@ -0,0 +1,173 @@
+ATTR_UPDATE = 'update'
+ATTR_AMOUNT = 'amount'
+ATTR_CHANGE = 'amount_change_30days'
+ATTR_CURRENCY = 'amount_currency'
+ATTR_LAST_UPDATED = 'amount_last_updated'
+ATTR_VAL_HI = 'valuation_range_high'
+ATTR_VAL_LOW = 'valuation_range_low'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_ZPID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Zestimate sensor."""
+    import xmltodict
+
+    name = config.get(CONF_NAME)
+    properties = config[CONF_ZPID]
+    params = {'zws-id': config[CONF_API_KEY]}
+
+    sensors = []
+    for zpid in properties:
+        params['zpid'] = zpid
+        try:
+            response = requests.get(_RESOURCE, params=params, timeout=5)
+            data = response.content.decode('utf-8')
+            data_dict = xmltodict.parse(data).get(ZESTIMATE)
+            error_code = int(data_dict['message']['code'])
+            if error_code != 0:
+                _LOGGER.error('The API returned: %s',
+                              data_dict['message']['text'])
+                return False
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error('The URL is not accessible')
+            return False
+
+        data = ZestimateData(params)
+        sensors.append(ZestimateDataSensor(name, data))
+    add_devices(sensors, True)
+
+
+class ZestimateDataSensor(Entity):
+    """Implementation of a Zestimate sensor."""
+
+    def __init__(self, name, data):
+        """Initialize the sensor."""
+        self.data = data
+        self._name = name
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        try:
+            return round(float(self._state), 1)
+        except ValueError:
+            return STATE_UNKNOWN
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {}
+        if self.data.measurings is not None:
+            if ATTR_AMOUNT in self.data.measurings:
+                data = self.data.measurings
+                attributes[ATTR_AMOUNT] = data[ATTR_AMOUNT]
+                attributes[ATTR_CURRENCY] = data[ATTR_CURRENCY]
+                attributes[ATTR_LAST_UPDATED] = data[ATTR_LAST_UPDATED]
+                attributes[ATTR_CHANGE] = data[ATTR_CHANGE]
+                attributes[ATTR_VAL_HI] = data[ATTR_VAL_HI]
+                attributes[ATTR_VAL_LOW] = data[ATTR_VAL_LOW]
+
+            attributes[ATTR_LOCATION] = self.data.address
+            attributes[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
+            return attributes
```

Comment:
Event though `None` is the default return value if your method doesn't return anything, I think it's nicer to still explicitly return `None` if there are no measurings. It's easier to read that way.